### PR TITLE
Add Serilog logging and hosting

### PR DESCRIPTION
**Pull request by KuraiAndras**
> The NuGet package needs to respect a few constraints in order to be listed in the curated list:
> - [X] Add a link to the NuGet package:
>  - https://www.nuget.org/packages/Serilog.Extensions.Hosting/4.2.0
> - [X] It must have non-preview versions (e.g 1.0.0 but not 1.0.0-preview.1)
> - [X] It must provide .NETStandard2.0 assemblies as part of its package
> - [X] The lowest version added must be the lowest .NETStandard2.0 version available
> - [X] The package has been tested with the Unity editor 
> - [X] The package has been tested with a Unity standalone player
>   - if the package is not compatible with standalone player, please add a comment to a Known issues section to the top level readme.md
> - [X] All package dependencies with .NETStandard 2.0 target must be added to the PR (respecting the same rules above)
>   - Note that if a future version of the package adds a new dependency, this dependency will have to be added manually as well

### CODE REVIEW

**Comment by nblumhardt** [comment]
Should be good to go now 👍

**Comment by bdovaz** [comment]
@nblumhardt UnityNuGet only allows NuGet package stable versions. I thought that you would publish a stable 3.1.0 version.

File: registry.json [hunk 1]
```diff
@@ -314,6 +314,14 @@
     "listed": true,
     "version": "2.9.0"
   },
+  "Serilog.Extensions.Hosting":{
+    "listed": true,
+    "version": "2.0.0"
+  },
+  "Serilog.Extensions.Logging":{
+    "listed": true,
+    "version": "2.0.0"
```
Comment (bdovaz) [inline registry.json:323]: Change to 3.1.0

**Review by bdovaz — CHANGES_REQUESTED** [review]

**Comment by bdovaz** [comment]
We need to create a PR to bump the hosting NuGet package version and depend on logging 3.1.0 to gully unblock this PR

**Comment by KuraiAndras** [comment]
@bdovaz I have created #69 so that the logging extension can be used before the hosting package is done

**Comment by KuraiAndras** [comment]
Once https://github.com/serilog/serilog-extensions-hosting/pull/55 is done, I can update the version of the Hosting package

**Comment by KuraiAndras** [comment]
@xoofx  @bdovaz  In the meanwhile .NET 6 came, and I think `Microsoft.Extensions.Hosting 6.0.0`  relies on `Microsoft.Extensions.Features` which is a new package. I also added that, since the build was failing

**Pull request merged**